nomad/job_endpoint.go

```diff
@@ -967,156 +967,188 @@ func (j *Job) BatchDeregister(args *structs.JobBatchDeregisterRequest, reply *st
 
 // Scale is used to modify one of the scaling targets in the job
 func (j *Job) Scale(args *structs.JobScaleRequest, reply *structs.JobRegisterResponse) error {
-	if done, err := j.srv.forward("Job.Scale", args, args, reply); done {
+	// Return early if request is successfully forwarded
+	if forwarded, err := j.srv.forward("Job.Scale", args, args, reply); forwarded {
 		return err
 	}
 	defer metrics.MeasureSince([]string{"nomad", "job", "scale"}, time.Now())
 
-	// Validate the arguments
-	namespace := args.Target[structs.ScalingTargetNamespace]
-	jobID := args.Target[structs.ScalingTargetJob]
-	groupName := args.Target[structs.ScalingTargetGroup]
-	if namespace != "" && namespace != args.RequestNamespace() {
-		return structs.NewErrRPCCoded(400, "namespace in payload did not match header")
-	} else if namespace == "" {
-		namespace = args.RequestNamespace()
-	}
-	if jobID != "" && jobID != args.JobID {
-		return fmt.Errorf("job ID in payload did not match URL")
-	}
-	if groupName == "" {
-		return structs.NewErrRPCCoded(400, "missing task group name for scaling action")
-	}
-	if args.Error && args.Count != nil {
-		return structs.NewErrRPCCoded(400, "scaling action should not contain count if error is true")
-	}
-	if args.Count != nil && *args.Count < 0 {
-		return structs.NewErrRPCCoded(400, "scaling action count can't be negative")
-	}
+	namespace := args.RequestNamespace()
```

Comment:
If it's not equivalent, an error is raised.

The ways it can go are:

1. You don't pass it, so RequestNamespace is used
2. You pass it, and it matches RequestNamespace, no error
3. You pass it, and it does not match RequestNamespace, which is an error

I don't believe this is a change in behavior.

Reply:
Yeah, reading the validation code more closely now: I agree - the validation didn't change and we always ensured these are equivalent, and use `RequestNamespace` when `Target[Namespace]` is empty.  This is cleaner indeed.  Thanks!